### ISSUE TITLE
tests failing in travis + python2

### ISSUE THREAD

**Issue by kad** [01org/bmap-tools#6]
```
Traceback (most recent call last):
  File "/home/travis/build/01org/bmap-tools/tests/test_filemap.py", line 152, in test
    _do_test(f_image, fiemap, mapped, unmapped)
  File "/home/travis/build/01org/bmap-tools/tests/test_filemap.py", line 115, in _do_test
    _check_ranges(f_image, filemap, first_block, blocks_cnt, mapped, "mapped")
  File "/home/travis/build/01org/bmap-tools/tests/test_filemap.py", line 90, in _check_ranges
    check[0], check[1]))
Error: bad or unmatching mapped range for file '/home/travis/build/01org/bmap-tools/sparse_7brm3p.img': correct is 18-18, get_mapped_ranges(0, 4096) returned 18-143
```

**Comment by kad** [comment]
seems to be build performed inside docker container over xfs filesystem.
reproducible on local xfs filesystem.

**Comment by kad** [comment]
a bit of investigation: on xfs speculative preallocation causing more blocks than needed to be mapped.
Solution seems to be to truncate file with overall destination image size before performing writes.
Speculative preallocation mostly kicks-in in cases where content is written after current EOF

**Comment by dedekind** [comment]
sounds like a good idea

Sent from my Android phone.

Best Regards,
Artem BItyutskiy

**Comment by kad** [comment]
Closed via #13
